topics/go/profiling/memcpu/stream.go

```diff
@@ -56,24 +56,27 @@ func assembleOutputStream() []byte {
 }
 
 func main() {
-	var output bytes.Buffer
+	var output1 bytes.Buffer
 	in := assembleInputStream()
 	out := assembleOutputStream()
 
 	find := []byte("elvis")
 	repl := []byte("Elvis")
 
 	fmt.Println("=======================================\nRunning Algorithm One")
-	output.Reset()
-	algOne(in, find, repl, &output)
-	matched := bytes.Compare(out, output.Bytes())
-	fmt.Printf("Matched: %v\nInp: [%s]\nExp: [%s]\nGot: [%s]\n", matched == 0, in, out, output.Bytes())
+	output1.Reset()
+	algOne(in, find, repl, &output1)
+	matched := bytes.Compare(out, output1.Bytes())
+	fmt.Printf("Matched: %v\nInp: [%s]\nExp: [%s]\nGot: [%s]\n", matched == 0, in, out, output1.Bytes())
 
 	fmt.Println("=======================================\nRunning Algorithm Two")
+	var output bytes.Buffer
 	output.Reset()
 	algTwo(in, find, repl, &output)
 	matched = bytes.Compare(out, output.Bytes())
 	fmt.Printf("Matched: %v\nInp: [%s]\nExp: [%s]\nGot: [%s]\n", matched == 0, in, out, output.Bytes())
+
+	fmt.Println("same output? ", bytes.Compare(output1.Bytes(), output.Bytes()))
```

Comment:
suggestion: it is a bit less easy on the eye maybe, but you could wrap these to blocks In empty `{}` and use the same variable name:

```
fmt.Println("=======================================\nRunning Algorithm One")
{
  var output bytes.Buffer
  ...
}

...

fmt.Println("=======================================\nRunning Algorithm Two")
{
    var output bytes.Buffer
    ...
}
```

here is an example on the playground: https://play.golang.org/p/lGvopALsPk7